tg_bot/modules/users.py

```diff
@@ -65,6 +68,136 @@ def broadcast(bot: Bot, update: Update):
+                                     html=True)
+
+                    except BadRequest as excp:
+                        if excp.message == "Chat not found":
+                            message.reply_text("Looks like I'm no longer a part of that chat!")
+                            return
+                        else:
+                            raise
+
+                else:
+                    message.reply_text("I'm already restricted from that chat!")
+
+            else:
+                message.reply_text("I can't seem to find the chat in my database. "
+                                   "Use /chatlist to obtain a list of chats in my database.")
+    
+        else:
+            message.reply_text("Invalid chat id! Make sure you include the '-' sign in the chat id.")
+
+    else:
+        message.reply_text("Incorrect number of arguments. Please use `/restrict chat_id`.",
+                           parse_mode = ParseMode.MARKDOWN)
+
+@run_async
+def new_member(bot: Bot, update: Update): # Leave group when added to restricted group
+    chat = update.effective_chat  # type: Optional[Chat]
+    new_members = update.effective_message.new_chat_members
+    user = update.effective_user
+
+    if sql.get_restriction(chat.id):
+        if not user.id in SUDO_USERS:
+            if any(new_mem.id == bot.id for new_mem in new_members):
+                update.effective_message.reply_text("I have been restricted by my admins from this chat! "
+                                                    "Request any of my admins to add me to this chat.")
+                bot.leave_chat(chat.id)
+
+        # Unrestrict group if a sudo user adds bot
+        else:
+            if any(new_mem.id == bot.id for new_mem in new_members):
+                sql.set_restriction(chat.id, restricted = False)
+
+                # Report to sudo users
+                send_to_list(bot, SUDO_USERS,
+                             "{} has added me to the chat <b>{}</b> and removed my restrictions."
+                             .format(mention_html(user.id, user.first_name), chat.title),
+                             html=True)
+
+@run_async
+def unrestrict_group(bot: Bot, update: Update, args: List[str]) -> str:
+    message = update.effective_message
+    
+    # Check if there is only one argument
+    if len(args) == 1:
+        chat_id = args[0]
+
+        # Check if chat_id is valid
+        if chat_id.startswith('-'):
+            # Check if chat_id is in bot database
+            all_chats = sql.get_all_chats() or []
+            if any(chat.chat_id == chat_id for chat in all_chats):
+                chat_restricted = sql.get_restriction(chat_id)
+                if chat_restricted:
+                    sql.set_restriction(chat_id, restricted = False)
+            
+                    message.reply_text("Successfully removed all restrictions on the chat with id `{}`"
```

Comment:
maybe add chat title too? to stay in line with the restrict function. make sure to escape it

Reply:
In the restrict function, I've used `chat_title = bot.get_chat(chat_id).title`, which won't work here because the bot is no longer in the chat. Should I use a new SQL function?